core/src/main/codegen/templates/Parser.jj

```diff
@@ -5484,6 +5484,10 @@ SqlNode BuiltinFunctionCall() :
     TimeUnit interval;
     final TimeUnit unit;
     final SqlNode node;
+<#if parser.allowTranslateUsingCharSetWithError>
+    boolean isWithError = false;
+    boolean allowTranslateUsingCharSet = false;
+</#if>
```

Comment:
Given that there are so many chunks being spliced into this function, I'd say it makes sense to extract the logic for the translate feature into its own function.

Reply:
It is a nice suggestion and I have think of the alternative. I will add more to the design doc and we can continue the discussion there.